server/eventHandlers/computeAPR.ts

```diff
@@ -0,0 +1,69 @@
+import { SubstrateTypes } from '@commonwealth/chain-events';
+import { sequelize } from '../database';
+import moment from 'moment';
+import BN from 'bn.js';
+
+
+const computeAPR = (commissionPer, rewardAmount, ownedAmount, totalStakeAmount, rewardsTimeIntervals) => {
+  /*
+    APR calculation for the current validator. Reference: https://github.com/hicommonwealth/commonwealth/blob/staking-ui/client/scripts/controllers/chain/substrate/staking.ts#L468-L472
+
+    commissionPer: Percentage (commission percentage is set by the validator and it will be in percent like 20%, 25.7%)
+    rewardAmount: Actual Reward amount awarded in the end of the session.
+    ownedAmount: Actual amount owned by the validator itself.
+    totalStakeAmount: Actual total stake amount which were staked during the session (It's is combination of Own Amount + nominator's staked amount).
+    rewardsTimeIntervals: Total Reward time intervals in seconds.
+  */
+  const rewardCommissionAmount = new BN(rewardAmount.toString()).muln(Number(commissionPer)).divn(100);;
+  const secondReward = ownedAmount.toBn().mul( (new BN(rewardAmount.toString())).sub(rewardCommissionAmount) ).div(totalStakeAmount.toBn() || new BN(1));
+  const totalReward = rewardCommissionAmount.add(secondReward);
+
+  const n = 1000000000; // Don't know what "n" is here.
+  const periodsInYear = (60 * 60 * 24 * 7 * 52) / rewardsTimeIntervals;
+  const percentage = (new BN(totalReward)).mul(new BN(n)).div(totalStakeAmount || new BN(1)).toNumber() / n;
+  const apr = percentage * periodsInYear;
+  return apr;
+};
+
+const getLast30DaysAPR = async(chain: String, event: String, stash: String ) => {
```

Comment:
rename the function to getAPR.
Pass the number of days as input argument. Tomorrow they might be computing apr over 90 days etc..